app/styles/modules/_password-row.scss

```diff
@@ -0,0 +1,118 @@
+/**
+ * There are some dependencies for passwords in the _input-row partial
+ */
+
+.password-row {
+  position: relative;
+
+  &.input-row .password {
+    padding-right: 75px;
+  }
+
+  .show-password {
+    opacity: 0;
+    position: absolute;
+  }
+
+  /**
+   * The addition of right, top, border-radius, and background-color are
+   * all to fix a bug in firefox 29 where the right padding of the password
+   * field is ignored and password text flows under the show/hide button. When
+   * Firefox is fixed, top and right can be set to 0, height set to 44px,
+   * border-radius and background-color removed.
+   *
+   * See https://bugzilla.mozilla.org/show_bug.cgi?id=965237
+   */
+
+  .show-password-label {
+    background-color: $content-background-color;
+    border-left: 1px solid $input-row-border-color;
+    color: $input-placeholder-color;
+    cursor: pointer;
+    font-size: $base-font;
+    min-width: 55px;
+    padding: 0 7px;
+    position: absolute;
+    right: 1px;
+    top: 1px;
+    // it is very easy to accidentally select the text when clicking
+    -webkit-touch-callout: none;
+    // user select not supported by autoprefixer
+    // https://github.com/postcss/autoprefixer/issues/167
+    -moz-user-select: none;
+    -ms-user-select: none;
+    -webkit-user-select: none;
+    user-select: none;
+    z-index: 3;
+
+    @include respond-to('big') {
+      border-radius: 0 $small-border-radius $small-border-radius 0;
+      height: 43px;
+      line-height: 42px;
+    }
+
+    @include respond-to('small') {
+      border-radius: 0;
+      height: 38px;
+      line-height: 38px;
+    }
+  }
+
+  .password:focus ~ .show-password-label {
+    border-left-color: $input-row-focus-border-color;
+  }
+
+  .password.invalid ~ .show-password-label {
+    border-left: 1px solid $error-background-color !important;
+  }
+
+}
+
+/**
+ * IE8 blows up when changing the type of the password field
```

Comment:
@johngruen - sounds good, mind filing an issue to remove IE8 support from CSS except for the "browsehappy" related stuff?